tools/traffic_vault_migrate/postgres.go

```diff
@@ -0,0 +1,716 @@
+package main
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import (
+	"database/sql"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/apache/trafficcontrol/lib/go-tc"
+	util "github.com/apache/trafficcontrol/lib/go-util"
+	_ "github.com/lib/pq"
+	"log"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+type PGConfig struct {
+	Host     string `json:"host"`
+	Port     string `json:"port"`
+	User     string `json:"user"`
+	Password string `json:"password"`
+	SSLMode  string `json:"sslmode"`
+	Database string `json:"database"`
+	Key      string `json:"aesKey"`
+	AESKey   []byte
+}
+type PGBackend struct {
+	sslKey PGSSLKeyTable
+	dnssec PGDNSSecTable
+	uri    PGURISignKeyTable
+	url    PGURLSigKeyTable
+	cfg    PGConfig
+	db     *sql.DB
+}
+
+func (pg *PGBackend) String() string {
+	data := fmt.Sprintf("PG server %v@%v:%v\n", pg.cfg.User, pg.cfg.Host, pg.cfg.Port)
+	data += fmt.Sprintf("\tSSL Keys: %v\n", len(pg.sslKey.Records))
+	data += fmt.Sprintf("\tDNSSec Keys: %v\n", len(pg.dnssec.Records))
+	data += fmt.Sprintf("\tURI Keys: %v\n", len(pg.uri.Records))
+	data += fmt.Sprintf("\tURL Keys: %v\n", len(pg.url.Records))
+	return data
+}
+func (pg *PGBackend) Name() string {
+	return "PG"
+}
+func (pg *PGBackend) ReadConfig(s string) error {
+	genericCfg, err := UnmarshalConfig(s, reflect.TypeOf(pg.cfg))
```

Comment:
The type is known; we don't need to use reflection here, do we?

Reply:
For code reuse, and consistency. Figured loading configs is something that should be done the same for each backend (Riak also calls this).

Reply:
It looks like the Riak backend calls it's own `ReadConfig` method that doesn't seem to be re-using this code - or vice-versa

Reply:
ahhh...I misread your earlier comment, nope there is no reason.